src/main/java/edu/ksu/canvas/model/Enrollment.java

```diff
@@ -237,4 +238,28 @@ public User getUser() {
     public void setUser(User user) {
         this.user = user;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Enrollment)) return false;
+        Enrollment that = (Enrollment) o;
+        // Use only attributes that can always be seen
+        return Objects.equals(courseId, that.courseId) &&
+                Objects.equals(courseSectionId, that.courseSectionId) &&
+                Objects.equals(enrollmentState, that.enrollmentState) &&
+                Objects.equals(limitPrivilegesToCourseSection, that.limitPrivilegesToCourseSection) &&
+                Objects.equals(rootAccountId, that.rootAccountId) &&
+                Objects.equals(type, that.type) &&
+                Objects.equals(userId, that.userId) &&
+                Objects.equals(associatedUserId, that.associatedUserId) &&
+                Objects.equals(role, that.role) &&
+                Objects.equals(user, that.user) &&
```

Comment:
User doesn't have equals/hashcode on it either, so this equals call will just check the memory address, which may not be equal.  Maybe you should consider implementing the same two methods on that class?

Reply:
Good call, I've done that too now.